### PR TITLE
CLI commands to check app and core signatures

### PR DESCRIPTION
**Pull request by VicDeo**

### CODE REVIEW

**Comment by VicDeo** [comment]
@LukasReschke I appreciate your :eyes:  on this

**Comment by LukasReschke** [comment]
Sure thing. Right on it.

**Comment by LukasReschke** [comment]
This works. However, if ownCloud is in a state where an update is required you might encounter output like:

```
➜  master git:(occ-check-signature) ✗ ./occ integrity:check-app files
ownCloud or one of the apps require upgrade - only a limited number of commands are available
You may use your browser or the occ upgrade command to do the upgrade
  - INVALID_HASH:
    - appinfo/info.xml:
      - expected: 567d2172d708be21e07b7dfdbb57ae78180fd5c1b07004444816cfcc381dccc48c7df8e5d88f2491e60f12f285c3659f6ef1065361ed045fb80d52c55dc858d0
      - current: fffb307b59d5f194b9c7cfb76a04ce95ad41a04865dd77b45ab207d6be5216de553ee9e331db18c9d729642aaf7f3977c81005ecdfe53df419f4caad47a89feb
```

So if you use that from the updater you probably need to account for that as well somehow.

**Comment by VicDeo** [comment]
@LukasReschke this is a common prefix for all commands. Unfortunately I haven't look into this yet. 
These two checks will be performed before migration anyway.

**Comment by VicDeo** [comment]
cc @DeepDiver1975 ^^

**Comment by LukasReschke** [comment]
:+1: if you use it like that :smile:

**Comment by LukasReschke** [comment]
> These two checks will be performed before migration anyway.

What might be also useful for you then is the `$path` function of `verifyAppSignature`. `verifyCore` doesn't have it yet though. (should be easy to add in any case)

In that case you could use the installed ownCloud's occ tool to check the downloaded files in the tmp directory by just passing a path. We do that also in the writing signature occ command :)

File: core/command/integrity/checkapp.php [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+ * @author Bart Visscher <bartv@thisnet.nl>
+ * @author Björn Schießle <schiessle@owncloud.com>
+ * @author Christian Kampka <christian@kampka.net>
+ * @author Joas Schilling <nickvergessen@owncloud.com>
+ * @author Lukas Reschke <lukas@owncloud.com>
+ * @author Morris Jobke <hey@morrisjobke.de>
+ * @author Robin McCorkell <robin@mccorkell.me.uk>
+ * @author Thomas Müller <thomas.mueller@tmit.eu>
+ * @author Victor Dubiniuk <dubiniuk@owncloud.com>
+ * @author Vincent Petry <pvince81@owncloud.com>
```
Comment (PVince81) [inline core/command/integrity/checkapp.php]: Did some many people collaborate when writing this new file ?
Looks like a copy-paste fail.

File: core/command/integrity/checkcore.php [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+ * @author Bart Visscher <bartv@thisnet.nl>
+ * @author Björn Schießle <schiessle@owncloud.com>
+ * @author Christian Kampka <christian@kampka.net>
+ * @author Joas Schilling <nickvergessen@owncloud.com>
+ * @author Lukas Reschke <lukas@owncloud.com>
+ * @author Morris Jobke <hey@morrisjobke.de>
+ * @author Robin McCorkell <robin@mccorkell.me.uk>
+ * @author Thomas Müller <thomas.mueller@tmit.eu>
+ * @author Victor Dubiniuk <dubiniuk@owncloud.com>
+ * @author Vincent Petry <pvince81@owncloud.com>
```
Comment (PVince81) [inline core/command/integrity/checkcore.php]: same here

**Comment by PVince81** [comment]
Code looks good otherwise :+1:

File: core/command/integrity/checkapp.php (hunk 1 above)
Comment (LukasReschke) [inline core/command/integrity/checkapp.php]: Sure. Did you forget the big meeting?

File: core/command/integrity/checkapp.php [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+
+/**
+ * Class CheckApp
+ *
+ * @package OC\Core\Command\Integrity
+ */
+class CheckApp extends Base {
+	public function __construct(Checker $checker) {
+		parent::__construct();
+		$this->checker = $checker;
```
Comment (MorrisJobke) [inline core/command/integrity/checkapp.php]: Please define this member variable.

File: core/command/integrity/checkcore.php [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+
+/**
+ * Class CheckCore
+ *
+ * @package OC\Core\Command\Integrity
+ */
+class CheckCore extends Base {
+	public function __construct(Checker $checker) {
+		parent::__construct();
+		$this->checker = $checker;
```
Comment (MorrisJobke) [inline core/command/integrity/checkcore.php]: Please define this member variable

**Comment by VicDeo** [comment]
@PVince I don't mind sharing a glory :smiley:  I have cut the list if you like it more.
@MorrisJobke Thanks, fixed.

**Pull request merged**